### PR TITLE
SANTUARIO-604: Use PSSParameterSpec for RSASSA-PSS without parameters SignatureMethod URIs

### PR DESCRIPTION
**Pull request by seanjmullan**
This change includes more portable support for the RSSSA-PSS algorithms without parameters defined in RFC 9231. 

It instantiates java.security.Signature objects with an "RSASSA-PSS" algorithm and initializes the default parameters using a `PSSParameterSpec` object. This approach works for both the JDK and BouncyCastle JCE providers. 

If this approach does not work for some reason, it falls back to instantiating Signature objects using algorithms like "SHA256withRSAandMGF1" (however these algorithm names are not standard as per the Java Security Standard Algorithm Names specification, so may not work for certain JCE providers, such as the ones in the JDK).

### CODE REVIEW

File: src/main/java/org/apache/jcp/xml/dsig/internal/dom/DOMSignatureMethod.java [hunk 1]
```diff
@@ -660,7 +712,12 @@ Type getAlgorithmType() {
         }
     }
 
-    static final class SHA1withRSAandMGF1 extends AbstractRSASignatureMethod {
+    static final class SHA1withRSAandMGF1 extends AbstractRSAPSSSignatureMethod {
+
+        private static PSSParameterSpec spec
```
[the remark this answers is not in the document]
Comment (seanjmullan) [reply src/main/java/org/apache/jcp/xml/dsig/internal/dom/DOMSignatureMethod.java]: Fixed.

File: src/main/java/org/apache/jcp/xml/dsig/internal/dom/DOMSignatureMethod.java [hunk 2]
```diff
@@ -362,6 +380,40 @@ byte[] preVerifyFormat(Key key, byte[] sig) {
...
+                Signature s = (p == null)
+                        ? Signature.getInstance("RSASSA-PSS")
+                        : Signature.getInstance("RSASSA-PSS", p);
+                try {
+                    s.setParameter(getPSSParameterSpec());
+                } catch (InvalidAlgorithmParameterException e) {
+                    throw new NoSuchAlgorithmException("Should not happen", e);
+                }
+                return s;
+            } catch (NoSuchAlgorithmException nsae) {
```
[the remark this answers is not in the document]
Comment (seanjmullan) [reply src/main/java/org/apache/jcp/xml/dsig/internal/dom/DOMSignatureMethod.java:415]: Fixed.

File: src/main/java/org/apache/jcp/xml/dsig/internal/dom/DOMSignatureMethod.java [hunk 3]
```diff
@@ -682,7 +743,12 @@ Type getAlgorithmType() {
         }
```
[the remark this answers is not in the document]
Comment (seanjmullan) [reply src/main/java/org/apache/jcp/xml/dsig/internal/dom/DOMSignatureMethod.java]: Good idea- fixed.

File: src/main/java/org/apache/xml/security/algorithms/implementations/SignatureBaseRSA.java [hunk 4]
```diff
@@ -364,10 +369,53 @@ public String engineGetURI() {
...
+        abstract PSSParameterSpec getPSSParameterSpec();
+    }
+
     /**
      * Class SignatureRSASHA1MGF1
      */
-    public static class SignatureRSASHA1MGF1 extends SignatureBaseRSA {
+    public static class SignatureRSASHA1MGF1 extends SignatureBaseRSAPSS {
+
+        private static PSSParameterSpec spec
```
[the remark this answers is not in the document]
Comment (seanjmullan) [reply src/main/java/org/apache/xml/security/algorithms/implementations/SignatureBaseRSA.java]: Fixed.

**Comment by coheigea** [comment]
@seanjmullan Reminder to backport to 3.0.x-fixes and 2.3.x-fixes

**Comment by seanjmullan** [comment]
> @seanjmullan Reminder to backport to 3.0.x-fixes and 2.3.x-fixes

Yes, thanks for the reminder.

**Pull request merged**